src/handlers.ts

```diff
@@ -46,7 +46,11 @@ export const takerRequestHandler = async (
 
     const response = await responsePromise;
     if (response && response.protocolVersion !== takerRequest.protocolVersion) {
-       throw new Error('Response and request protocol versions do not match');
+        console.error('Response and request protocol versions do not match');
+        return res
+            .status(HttpStatus.NOT_IMPLEMENTED)
+            .json({ errors: ['Server does not support the requested protocol version'] })
```

Comment:
This doesn't seem to be the right error message

Reply:
I suppose this is fine, it's saying the server doesn't correctly handle the versioned request